src/jit/lsra.h

```diff
@@ -837,6 +837,51 @@ class LinearScan : public LinearScanInterface
 
     unsigned getWeight(RefPosition* refPos);
 
+    // Map from a reg optional marked ref position to its consuming tree node.
+    // This info is set up while creating reg optional ref positions and used
+    // to determine that srcMemOpCount of consuming tree node is not
+    // exceeded while allocating reg optional use positions of its operands.
+    typedef SimplerHashTable<RefPosition*, PtrKeyFuncs<RefPosition>, GenTree*, JitSimplerHashBehavior>
```

Comment:
@CarolEidt's suggestion would reduce the number of hastable lookups in allocation path if we implement a simple model where there is never more than two operands marked as reg-optional and at the most only one of them can be in memory.  This model would meet xarch needs.  Note that the simple model will use the same number of hash table entries as my current implementation.

I have implemented this suggestion and would like to mention a couple of things:
1) Now LSRA will have logic in buildRefPositionsForNode() and allocateRegisters() that depends on the above mentioned assumptions.
2) Also, I could get rid of srcMemOpCount altogether.  I have replaced it by srcRegOptionalCount in debug only builds to assert that no more than two operands of a node are marked as reg optional.